core/src/test/scala/akka/persistence/r2dbc/TestConfig.scala

```diff
@@ -35,6 +35,23 @@ object TestConfig {
             database = "yugabyte"
           }
           """)
+      case "h2" =>
+        ConfigFactory.parseString("""
+          akka.persistence.r2dbc.connection-factory {
+           driver = "h2"
+           #url = "r2dbc:h2:mem:///test-db;DB_CLOSE_DELAY=-1"
+           url = "r2dbc:h2:file:///./target/h2-test-db;DB_CLOSE_DELAY=-1"
```

Comment:
instead of this I think we also need dialect specific implementations of ConnectionFactoryProvider, which is using the builder api

Reply:
I need to think about the config/API for user and how we deal with it internally for the connection factory.

Feels a bit like it should be tied to dialect in config file somehow, so that you never mix the dialect and type of connection config up, but we'd want the existing config to keep working ofc. Hmmmm.

Reply:
One idea is to default to a `akka.persistence.r2dbc.[dialect].connection-factory` and make the Postgres one inherit the current `connection-factory` block (and deprecate that).

Reply:
Note that most properties are not specific to a dialect. Those are defined in `io.r2dbc.spi.ConnectionFactoryOptions`. I don't know if that influence the decision, but just saying.

Reply:
Since there is no support for separate-process-H2, none of the host, port, user, pwd etc are really applicable. Ofc we could parse all the ones from `ConnectionFactoryOptions` and just ignore the irrelevant ones for H2.

Not pushed yet, but in my current direction I have tried to break out the connection pool ones, which are completely detached from the dialect/connection factory settings, into a separate class and made the existing `ConnectionFactorySettings` an internal detail of the Postgres:y dialects, possibly dropping it completely and just parsing config into the r2dbc builder.

Reply:
Pushed that restructure work in 1e31c54e7b97bf5d724bbc2be5526d9e2db088fe